plugins/modules/role.py

```diff
@@ -75,16 +74,21 @@
 
 from ansible.module_utils.basic import AnsibleModule
 from ansible_collections.sensu.sensu_go.plugins.module_utils import (
-    arguments, errors, utils
+    arguments, errors, utils, role_utils
 )
 
 
+def validate_module_params(module):
+    params = module.params
+    if params['state'] == 'present':
+        if not params['rules']:
+            module.fail_json(
+                msg='state is present but all of the following are missing: rules'
+            )
```

Comment:
As you have already said, it would definitely make sense to move this check into common utility module. For example, we could add a method like this:

```
def validate_module_params(params):
    if params['state'] == 'present':
        if not params['rules']:
            return 'state is present but all of the following are missing: rules'
    return None
```

And then call it like

```
if msg := role_utils.validate_module_params(module.params):
    module.fail_json(msg=msg)
```

This way we can keep the utility functions pure and at least one developer happy ;)

Reply:
Done, albeit without the `:=` operator